### PR TITLE
setup to pyproject.toml

### PR DESCRIPTION
**Pull request by guillemcortes**
Still WIP, but this PR addresses #158 by switching the project setup to pyproject.toml.

### CODE REVIEW

**Comment by guillemcortes** [comment]
Hi, I've tested this setup by installing the dependencies in a new environment with
```
pip install .
pip install ."[tests]"
pip install ."[docs]"
```

And it works fine. 

For setting the soundata version, I've tried to modify the pipeline as little as possible. In the packaging.python guidelines https://packaging.python.org/en/latest/guides/single-sourcing-package-version/ they state that can be done via multiple ways. Maybe it is worth exploring them in the future.

Could you review it, @genisplaja? thanks!

**Comment by guillemcortes** [comment]
Hi @hadware @hagenw,

We've been working on this PR that addresses #153 #154 and #158. We've added a .toml file and reviewed the necessary dependencies. We've also marked the packages required for the plotting functionalities as optional so they can be installed via `pip install soundata[plots]`. You can find more info about these plotting functionalities in #171 (which we plan to resolve once this PR is merged). Let us know if this PR would resolve all your concerns or if there is something that we have overlooked. Many thanks!

File: pyproject.toml
Comment (hagenw) [inline pyproject.toml]: `mypy` is not imported by the package, but only mentioned in `docs/source/contributing.rst`, e.g.
```rst
python -m mypy soundata --ignore-missing-imports --allow-subclassing-any
```
So I guess this can be moved to optional dependencies.

**Comment by hagenw** [comment]
Thanks, this indeed addresses my concerns from #153 and #154.

**Comment by hadware** [comment]
Could you address the issue regarding mypy and flake8? Otherwise this is good.

File: pyproject.toml
Comment (hagenw) [inline pyproject.toml:64]: You are recommending to install the `docs` requirements via this file. I guess you can then remove `docs/requirements.txt` altogether?

Comment (genisplaja) [reply pyproject.toml:64]: This file is currently used here: [readthedocs.yml](https://github.com/soundata/soundata/blob/5e7aebcfb5b048bfdb2f8a91fe19b4c9a3b2738c/.readthedocs.yml#L26). Maybe we can remove the whole optional python installation instructions is this file since contributors will be already installing the required documentation dependencies through pyproject if they want to contribute (although we might need to document that further in the documentation)? Then, we would be able to remove the docs/requirements.txt file.

Comment (guillemcortes) [reply pyproject.toml:64]: I've found this https://docs.readthedocs.io/en/stable/config-file/v2.html#packages so according to it we should be able to specify the requirements with .toml file and specify the readthedocs requirements with:
```yaml
python:
  install:
    - method: pip
      path: .
      extra_requirements:
        - docs
```
that will run `pip install .[docs]`

Comment (genisplaja) [reply pyproject.toml:64]: neat!

Comment (hagenw) [reply pyproject.toml:64]: Thanks, now we have a single source of truth for the documentation requirements.

File: pyproject.toml
Comment (hagenw) [inline pyproject.toml:13]: The list of authors here does not match the list of authors you selected for the paper.

**Comment by guillemcortes** [comment]
Hi @hagenw ,

I added the full list of authors to match those of the JOSS article and also the maintainers. I have decided to only include Magdalena's email since she is the corresponding author. I've also clarified the project.urls so users can quickly check how to open an issue in Soundata.

File: setup.py
Comment (guillemcortes) [inline setup.py:13]: The idea is to host all the indexes remotely, so I think we'll have to remove this line, right? Although I think it's better if we leave this PR as is and change it in https://github.com/soundata/soundata/pull/169 directly. @genisplaja @hagenw

Comment (hagenw) [reply setup.py:13]: Yes, I would tackle that in a separate pull request e.g. #169) to not mix too many things.

**Review by hagenw — APPROVED** [review]

**Review by genisplaja — APPROVED** [review]
That's ready to go :)

**Pull request merged**
